### PR TITLE
SP2-855 design review remediation

### PR DESCRIPTION
**Pull request by alexw-moj**
Updating a future goal will now return to the the future goals tab.

Reorders the steps if they have been updated.

‘Update’ > ‘Add or change steps’ > ‘Save and continue’ - goes back to the update page.

More space added between the 'add or change steps' link and the 'add notes' title

### CODE REVIEW

File: server/utils/utils.ts [hunk 1]
```diff
@@ -119,3 +121,27 @@ export function mergeDeep(...objects: Record<string, any>[]): Record<string, any
     return newObj
   }, {})
 }
+
+/*
+  Depending on status, steps should be sorted by status, with the ones that have been updated being moved to the bottom of that sorted section.
+  See the Statuses - in order.png for expected ordering.
```
Comment (pipwilson) [inline server/utils/utils.ts]: maybe a link to the jira ticket or figma would be better than just the image filename?

File: server/routes/add-steps/AddStepsController.ts [hunk 2]
```diff
@@ -49,7 +49,10 @@ export default class AddStepsController {
       goalUuid,
     )
 
-    return res.redirect(`${URLs.PLAN_OVERVIEW}?status=success`)
+    const link = req.session.backLink ? req.session.backLink : `${URLs.PLAN_OVERVIEW}?status=success`
+    req.session.backLink = null
```
Comment (Thomas-Geraghty) [inline server/routes/add-steps/AddStepsController.ts]: Maybe move this into a sessionService function?

File: server/utils/utils.ts (hunk 1 above)
Comment (Thomas-Geraghty) [reply server/utils/utils.ts]: I think there might be concerns about posting links to internal jira/confluence/tooling on the public repos. If the comment is clear enough, we should be okay here 😄

**Comment by Thomas-Geraghty** [comment]
Looks good, just remove the cute little submit button that appears under the 'Review steps' text 🥳 

update: Resolved in https://github.com/ministryofjustice/hmpps-sentence-plan-ui/pull/178/commits/e7af8c4393e43a9bb118d5b4854cb06e7ce2bb8b

File: server/utils/utils.ts (hunk 1 above)
Comment (pipwilson) [reply server/utils/utils.ts]: the ID of the jira ticket would be enough here - something to give an idea of where the referenced image is.

Comment (alexw-moj) [reply server/utils/utils.ts]: Updated in https://github.com/ministryofjustice/hmpps-sentence-plan-ui/pull/178/commits/285981fb2a55dad2eb089f28cb993c126cde65c3

File: server/routes/add-steps/AddStepsController.ts [hunk 3]
```diff
@@ -49,8 +49,10 @@ export default class AddStepsController {
       goalUuid,
     )
 
-    const link = req.session.backLink ? req.session.backLink : `${URLs.PLAN_OVERVIEW}?status=success`
-    req.session.backLink = null
+    const link = req.services.sessionService.getBackLink()
+      ? req.services.sessionService.getBackLink()
+      : `${URLs.PLAN_OVERVIEW}?status=success`
```
Comment (Thomas-Geraghty) [inline server/routes/add-steps/AddStepsController.ts]:
```
   const link = req.services.sessionService.getBackLink() 
      ?? `${URLs.PLAN_OVERVIEW}?status=success`;
 ```
would be cleaner 😃

Comment (alexw-moj) [reply server/routes/add-steps/AddStepsController.ts]: Resolved in https://github.com/ministryofjustice/hmpps-sentence-plan-ui/pull/178/commits/f64b266e0b5f770ab7c1355ae4da5a3fdd569630

File: server/routes/add-steps/AddStepsController.ts (hunk 2 above)
Comment (alexw-moj) [reply server/routes/add-steps/AddStepsController.ts]: Done in https://github.com/ministryofjustice/hmpps-sentence-plan-ui/pull/178/commits/f457ddb2ae29af2d32e4a40e0db271066c1028b5

**Review by pipwilson — DISMISSED** [review]
LGTM

**Review by Thomas-Geraghty — DISMISSED** [review]

**Review by pipwilson — APPROVED** [review]
🚢

**Pull request merged**